packages/components/src/InputTime/InputTime.rebuilt.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { ChangeEvent, useId, useRef } from "react";
+import { useTimePredict } from "./hooks/useTimePredict";
+import { InputTimeProps, InputTimeRebuiltProps } from "./InputTime.Types";
+import { dateToTimeString, timeStringToDate } from "./utils/input-time-utils";
+import { FormFieldWrapper, useFormFieldWrapperStyles } from "../FormField";
+
+export function InputTimeRebuilt({
+  defaultValue,
+  value,
+  onChange,
+  ...params
+}: InputTimeRebuiltProps) {
+  const ref =
+    (params.inputRef as React.RefObject<HTMLInputElement>) ??
+    useRef<HTMLInputElement>(null);
+  const { setTypedTime } = useTimePredict({
+    value,
+    handleChange,
+  });
+
+  const { inputStyle } = useFormFieldWrapperStyles(params);
+
+  const id = getId(params);
+
+  const valueProps = {
+    ...(defaultValue && { defaultValue: dateToTimeString(defaultValue) }),
+    ...(!defaultValue && { value: dateToTimeString(value) }),
+  };
+
+  return (
+    <FormFieldWrapper
+      disabled={params.disabled}
+      size={params.size}
+      align={params.align}
+      inline={params.inline}
+      name={params.name}
+      error={""}
+      identifier={id}
+      descriptionIdentifier={`descriptionUUID--${id}`}
+      invalid={Boolean(params.invalid)}
+      description={params.description}
+      clearable={params.clearable ?? "never"}
+      onClear={handleClear}
+      placeholder={params.placeholder}
+      value={value}
+    >
+      <input
+        ref={ref}
```

Comment:
The rebuilt version can no longer be disabled

Reply:
ah yes, the disabled is only being passed to the `FormFieldWrapper` but we need it on the input itself. good catch.